### ISSUE TITLE
hoiquylogitics

### ISSUE THREAD

**Issue by hieuhaihuoc** [hieuhaihuoc/Finding-Athorigum#2]
import numpy as np
import matplotlib.pyplot as plt
from sklearn.preprocessing import PolynomialFeatures
from sklearn.linear_model import LinearRegression

# Cố định seed để kết quả không thay đổi mỗi lần chạy
np.random.seed(42)

# Tạo đữ liệu
num_samples = 100
X = np.random.rand(num_samples, 1) * 6 - 3
noise = np.random.randn(num_samples, 1)

# Hàm thật: y = 0.5x^2 + x + 2 + noise
y = 0.5 * X**2 + X + 2 + noise

# biến đổi đa thức bậc 2
poly = PolynomialFeatures(degree=2, include_bias=False)
X_poly = poly.fit_transform(X)

print("X ban đầu:", X[0])
print("Sau biến đổi [x, x^2]:", X_poly[0])

# ===== HUẤN LUYỆN LINEAR REGRESSION =====
model = LinearRegression()
model.fit(X_poly, y)

print("\nKết quả huấn luyện:")
print("Intercept:", model.intercept_)
print("Coefficients:", model.coef_)

# ===== DỰ ĐOÁN =====
X_test = np.linspace(-3, 3, 100).reshape(-1, 1)
X_test_poly = poly.transform(X_test)
y_pred = model.predict(X_test_poly)

'''
# ===== VẼ ĐỒ THỊ =====
plt.figure(figsize=(9, 5))
plt.scatter(X, y, label="Dữ liệu thực tế")
plt.plot(X_test, y_pred, linewidth=2, label="Polynomial Regression")
plt.grid(True, alpha=0.3)
plt.xlabel("X")
plt.ylabel("y")
plt.title("Polynomial Regression")
plt.legend()
plt.show()
'''

# Đồ thị
plt.figure(figsize=(9, 5))
plt.scatter(X, y, label="Dữ liệu thực tế")
plt.plot(X_test, y_pred, linewidth=2, label="Polynomial Regression")
plt.grid(True, alpha=0.3)
plt.xlabel("X")
plt.ylabel("y")
plt.title("Polynomial Regression")
plt.legend()

# lưu ảnh
plt.savefig("polynomial_regression.png", dpi=150)
print("Đã lưu hình: polynomial_regression.png")
